webxr/headless/mod.rs

```diff
@@ -39,12 +46,22 @@ struct HeadlessDiscovery {
     receiver: Option<Receiver<MockDeviceMsg>>,
 }
 
+struct InputInfo {
+    source: InputSource,
+    active: bool,
+    pointer: TypedRigidTransform3D<f32, Input, Native>,
+}
+
 struct HeadlessDevice {
     gl: Rc<dyn Gl>,
     floor_transform: TypedRigidTransform3D<f32, Native, Floor>,
     viewer_origin: TypedRigidTransform3D<f32, Viewer, Native>,
     views: Views,
     receiver: Receiver<MockDeviceMsg>,
+    events: EventBuffer,
+    inputs: Vec<InputInfo>,
+    disconnect_callbacks: Vec<Sender<()>>,
```

Comment:
Why is disconnect treated differently from the other events?

Reply:
The mock API requires we signal disconnection to have succeeded.